algorithms/linfa-ica/src/error.rs

```diff
@@ -18,9 +18,12 @@ pub enum FastIcaError {
     SvdDecomposition,
     #[error("tolerance should be positive but is {0}")]
     InvalidTolerance(f32),
+    #[cfg(feature = "blas")]
+    #[error("Linalg BLAS error: {0}")]
+    LinalgBlasError(#[from] ndarray_linalg::error::LinalgError),
+    #[error("Linalg error: {0}")]
     /// Errors encountered during linear algebra operations
-    #[error("Linalg Error: {0}")]
-    Linalg(#[from] LinalgError),
+    LinalgError(#[from] linfa_linalg::LinalgError),
```

Comment:
`#[cfg(not(feature = "blas"))]` missing before this one or it is not really a problem?

Reply:
It's supposed to be absent, to make the `blas` feature strictly additive. If I gated `LinalgError` on the absence of `blas`, then code that depends on the non-`blas` version would break if `blas` was enabled anywhere else in the dependency tree.